csrc/scheduler/matmul_heuristic_plugin_api.h

```diff
@@ -0,0 +1,90 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <memory>
+
+namespace nvfuser {
+
+namespace matmul_heuristic_plugin {
+
+//! This is intended as a minimal interface for enabling matmul heuristics.
+//! In order to plug in your own custom heuristic, create a dynamic library
+//! defining a subclass of KernelConfig, overriding the `configure` method. This
+//! class does not need to be exported from the dll, but you should export a
+//! std::unique_ptr<KernelConfig> makeConfig() function that returns a
+//! unique_ptr to an object of that type. The `configure` method will be called
+//! on that object by nvfuser in order to fill the correct values in the class.
+//!
+//! If that library is located at /path/to/libfoo.so you can set
+//! NVFUSER_MATMUL_HEURISTIC_PLUGIN=/path/to/libfoo.so to use the plugin to
+//! determine matmul parameters automatically.
+
+//! This is the return type of a HeuristicFunc (defined below) implemented in a
+//! plugin. This is used to set values in MatmulParams
+struct KernelConfig {
```

Comment:
Possibly related doc for subclassing a C++ class in python:

https://pybind11.readthedocs.io/en/stable/advanced/classes.html